nixos/maintainers/scripts/lxd/lxd-image.nix

```diff
@@ -0,0 +1,34 @@
+{ lib, config, pkgs, ... }:
+
+with lib;
+
+{
+  imports = [
+    ../../../modules/virtualisation/lxc-container.nix
+  ];
+
+  virtualisation.lxc.templates.nix = {
+    enable = true;
+    target = "/etc/nixos/lxd.nix";
+    template = ./nix.tpl;
+    when = [ "create" "copy" ];
+  };
+
+  # copy the config for nixos-rebuild
+  system.activationScripts.config = ''
+    if [ ! -e /etc/nixos/configuration.nix ]; then
+      mkdir -p /etc/nixos
+      cat ${./lxd-image-inner.nix} > /etc/nixos/configuration.nix
+    fi
+  '';
+
+  # Make lxc exec work properly
```

Comment:
can lxc fall back to a `/bin/sh` for `lxc exec` if it doesn't find `/bin/bash`? That'd probably even be applicable upstream.

Reply:
The problem is with the sh not having the path set properly actually. Plus a reduced developer experience because of that and sh's rather minimal featureset.

Reply:
Well, `/bin/sh` IS a symlink to `bash-interactive` on NixOS, so there's no worse developer experience. And even for non-NixOS, giving a stripped down `/bin/sh` instead of failing entirely might be preferrable.

Reply:
Please note that while /bin/sh is indeed a symlink to bash, bash was programmed to behave like sh when argv[0] is sh (when the link is called "sh"). Meaning it will only ever have the features posix sh has enabled, unless it gets called as /bin/bash (or literally anything that is not sh). If it doesn't load the profile then that is a problem.

Ofc if you know of a way to get sh to load the profile in this particular circumstance I'm all ears. Just without $PATH set correctly the shell isn't exactly useful